marklogic-data-hub/src/test/java/com/marklogic/hub/dhs/installer/command/CanInstallDhsCommandTest.java

```diff
@@ -1,53 +1,50 @@
 package com.marklogic.hub.dhs.installer.command;
 
 import com.fasterxml.jackson.databind.node.ObjectNode;
-import com.marklogic.hub.AbstractHubCoreTest;
-import com.marklogic.hub.impl.Versions;
+import com.marklogic.hub.impl.MarkLogicVersion;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
 
-class CanInstallDhsCommandTest extends AbstractHubCoreTest {
+class CanInstallDhsCommandTest {
 
-    private InstallIntoDhsCommand installIntoDhsCommand = new InstallIntoDhsCommand();
-
-    @Autowired
-    Versions versions;
+    private final InstallIntoDhsCommand installIntoDhsCommand = new InstallIntoDhsCommand();
 
     @Test
     void testCanInstallDhs() {
-        canInstall(installIntoDhsCommand.canInstallDhs("5.0.2", versions.getMLVersion("10.0-3")));
-        canInstall(installIntoDhsCommand.canInstallDhs("5.1.0", versions.getMLVersion("10.0-4.2")));
-        canInstall(installIntoDhsCommand.canInstallDhs("5.2.0", versions.getMLVersion("10.0-5")));
-        canInstall(installIntoDhsCommand.canInstallDhs("5.2.3", versions.getMLVersion("10.0-3.1")));
-        canInstall(installIntoDhsCommand.canInstallDhs("5.0.0", versions.getMLVersion("10.0-3")));
-
-        cantInstallInvalidDhfVersion(installIntoDhsCommand.canInstallDhs("4.0.2", versions.getMLVersion("10.0-3")));
-        cantInstallInvalidDhfVersion(installIntoDhsCommand.canInstallDhs("4.3.2", versions.getMLVersion("10.0-4.2")));
-        cantInstallInvalidDhfVersion(installIntoDhsCommand.canInstallDhs("3.0.0", versions.getMLVersion("10.0-5")));
-        cantInstallInvalidDhfVersion(installIntoDhsCommand.canInstallDhs("2.3.2", versions.getMLVersion("9.0.0")));
-        cantInstallInvalidDhfVersion(installIntoDhsCommand.canInstallDhs("4.1.0", versions.getMLVersion("9.0-12")));
-
-        cantInstallInvalidServerVersion(installIntoDhsCommand.canInstallDhs("5.0.2", versions.getMLVersion("9.0-12")));
-        cantInstallInvalidServerVersion(installIntoDhsCommand.canInstallDhs("5.1.0", versions.getMLVersion("10.0-2.4")));
-        cantInstallInvalidServerVersion(installIntoDhsCommand.canInstallDhs("5.2.0", versions.getMLVersion("9.0-15")));
-        cantInstallInvalidServerVersion(installIntoDhsCommand.canInstallDhs("5.2.3", versions.getMLVersion("10.0-2")));
-        cantInstallInvalidServerVersion(installIntoDhsCommand.canInstallDhs("5.0.0", versions.getMLVersion("10.0-2.12")));
-
-        canInstall(installIntoDhsCommand.canInstallDhs(null, versions.getMLVersion("10.0-3")));
-        canInstall(installIntoDhsCommand.canInstallDhs(null, versions.getMLVersion("11.0-1")));
-        cantInstallInvalidServerVersion(installIntoDhsCommand.canInstallDhs(null, versions.getMLVersion("9.0.0")));
-        cantInstallInvalidServerVersion(installIntoDhsCommand.canInstallDhs(null, versions.getMLVersion("9.0-12")));
+        canInstall(installIntoDhsCommand.canInstallDhs("5.0.2", new MarkLogicVersion("10.0-3")));
```

Comment:
A testing tip here is that when you have to call the same method/constructor repeatedly in a test, it's a good idea to make a little private method that does that, and then have the test call that private method.

The benefit is that if I search on references for this constructor, I get dozens of them, which makes it harder to understand where the constructor is used. With a private method calling the constructor in the test, I'll just get a handful of them.

Reply:
Makes sense; added a private method for creating the object.